pkg/tasks/metricsserver.go

```diff
@@ -129,20 +129,15 @@ func (t *MetricsServerTask) create(ctx context.Context) error {
 		if err != nil {
 			return err
 		}
-
-		mcs, err := t.factory.MetricsClientCerts()
+		metricsCert, err := t.factory.MetricsServerClientCerts()
 		if err != nil {
-			return fmt.Errorf("initializing metrics-client-cert failed: %w", err)
+			return fmt.Errorf("initializing Metrics Server Client Certs secret failed: %w", err)
 		}
 
-		mcs, err = t.client.WaitForSecret(
-			ctx,
-			mcs,
-		)
+		metricsServerClientCert, err := t.client.WaitForSecret(ctx, metricsCert)
```

Comment:
we can use the same `metricsCert` to avoid adding another variable, there are enough already :)